core/app/components/MemberSelect/MemberSelectClient.tsx

```diff
@@ -96,8 +96,13 @@ export function MemberSelectClient({
 
 	const onInputValueChange = useDebouncedCallback((value: string) => {
 		const newParams = new URLSearchParams(params);
+		const oldParams = newParams.toString();
 		newParams.set(queryParamName, value);
-		router.replace(`${pathname}?${newParams.toString()}`, { scroll: false });
+		// Only change params when they are different, otherwise can cause race conditions
```

Comment:
this is only a workaround, and I'm not sure what the 'proper' solution is. basically it seems like this `onInputValueChange` was triggering on form submit, and therefore overwriting the parameters I was trying to set on form submit (namely, that the the form was completed). instead, this seemed to trigger, and there'd be a race condition in setting the URL params, so sometimes the form would never get marked as completed.

I don't know why this func is triggered when the form submits, so the workaround here is just to not set the new params if they aren't different. It seems like the deeper problem might be that this func is firing too often though